frontend/src/components/Layout/AppBar/Buttons/TaskManagerButton.vue

```diff
@@ -70,104 +68,57 @@ const props = withDefaults(
 );
 
 const menu = ref(false);
-const taskList = ref<TaskInfo[]>([]);
-const scheduledTimeout = ref(0);
 const taskManager = taskManagerStore();
-
-const buttonColor = computed(() => {
-  return taskList.value.every((task) => {
-    return task.progress === 100;
-  })
-    ? 'success'
-    : undefined;
-});
-
-const showButton = computed(() => taskList.value.length > 0);
-
-/**
- *  Clear the tasks list
- */
-function clearState(): void {
-  taskList.value = [];
-}
-
-/**
- * Set a timeout to hide the button
- */
-function setTimeout(): void {
-  scheduledTimeout.value = window.setTimeout(clearState, props.timeout);
-}
-
-/**
- * Locally retrieve and parse the tasks list
- */
-function getTaskList(): void {
-  const list: Array<TaskInfo> = [];
-
-  for (const task of taskManager.tasks) {
-    switch (task.type) {
+let frozenTaskList: readonly TaskInfo[] = [];
+const runningTaskList = computed<TaskInfo[]>(() => {
+  return taskManager.tasks.map((t) => {
+    switch (t.type) {
       case TaskType.ConfigSync: {
-        list.push({
-          progress: undefined,
+        return {
+          progress: t.progress,
           textKey: 'appbar.tasks.configSync',
-          id: task.id
-        });
-
-        break;
+          id: t.id
+        };
       }
       case TaskType.LibraryRefresh: {
-        list.push({
-          progress: task.progress,
+        return {
+          progress: t.progress,
           textKey: 'appbar.tasks.scanningLibrary',
           textParams: {
-            library: task.data || ''
+            library: t.data || ''
           },
-          id: task.id
-        });
-
-        break;
+          id: t.id
+        };
       }
     }
-  }
+  });
+});
 
-  const taskIds = new Set(
-    list.map((task) => {
-      return task.id;
-    })
-  );
-  const finishedTasks: Array<TaskInfo> = [];
+const allCompleted = computed(() =>
+  runningTaskList.value.every((t) => t.progress === 100)
+);
 
-  for (const task of taskList.value) {
-    if (!taskIds.has(task.id)) {
-      task.progress = 100;
-      finishedTasks.push(task);
-    }
-  }
+const buttonColor = computed(() => {
+  return allCompleted.value ? 'success' : undefined;
+});
 
-  taskList.value = [...list, ...finishedTasks];
-}
+const taskList = computed(() => {
+  return menu.value && allCompleted.value
+    ? frozenTaskList
+    : runningTaskList.value;
+});
+
+const showButton = computed(() => taskList.value.length > 0);
 
 watch(
-  taskManager.tasks,
-  () => {
-    getTaskList();
-  },
+  () => props.timeout,
+  () => (taskManager.timeout = props.timeout),
```

Comment:
@aweebs I doubted as well with how to approach it, but after your comment I'm probably going to remove it.

As I did with the concern separation of the websocket store (moving all the watchers to the relevant places to what they do), it looked strange for me to leave all the control of an UX-exclusive feature in the store, instead of the component itself.

I'm going to remove it because yeah, it doesn't make much sense either way, so it's a tradeoff we need to take (because all the task cleanup logic must live in the store ofc).

I think you don't understand very well what this timeout does though. I'll describe it in your following comments.